keps/976-plain-pods/README.md

```diff
@@ -0,0 +1,678 @@
+
+#### Pods subject to queueing
+
+Not all Pods in a cluster should be subject to queueing.
+In particular the following pods should be excluded from getting the scheduling gate or label.
+
+1. Pods owned by other job APIs managed by kueue.
+
+They can be identified by the ownerReference, based on the list of enabled integrations.
+
+In some scenarios, users might have custom job objects that own Pods through an indirect object.
+In these cases, it might be simpler to identify the pods through a label selector.
+
+2. Pods belonging to specific namespaces (such as kube-system or kueue-system).
+
+The namespaces and pod selectors are defined in Configuration.Integrations.
+For a Pod to qualify for queueing by Kueue, it needs to satisfy both the namespace and pod selector.
+
+```golang
+type Integrations struct {
+  Frameworks []string
+  PodOptions *PodIntegrationPolicy
+}
+
+type PodIntegrationOptions struct {
+  NamespaceSelector *metav1.LabelSelector
+  PodSelector *metav1.LabelSelector
+}
+```
+
+When empty, Kueue uses the following NamespaceSelector internally:
+
+```yaml
+matchExpressions:
+- key: kubernetes.io/metadata.name
+  operator: NotIn
+  values: [kube-system, kueue-system]
+```
+
+### Constructing Workload objects
+
+Once the webhook has marked Pods subject to queuing with the `kueue.x-k8s.io/managed: true` label,
+the Pod reconciler can create the corresponding Workload object to feed the Kueue admission logic.
+
+Note that the Workload cannot be owned by the Pod. Otherwise any cascade deletion of the Pod would
+delete the Workload object, even before the Pod terminates (if it has a grace period).
+This means that the controller needs to manually delete the Workload object once it has the
+Finished condition (after we have determined that all pods have finished).
+
+<<[UNRESOLVED TTL after finished]>>
+A possible extension here is to add a TTL, but we will get user feedback first.
+<<[/UNRESOLVED]>>
+
+#### Single Pods
+
+The simplest case we want to support is single Pod jobs. These Pods only have the label
+`kueue.x-k8s.io/queue-name`, indicating the local queue where they will be queued.
+
+The Workload for the Pod in [story 1](#story-1) would look as follows:
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: Workload
+metadata:
+  name: pod-foo
+  namespace: pod-namespace
+spec:
+  queueName: queue-name
+  podSets:
+  - count: 1
+    name: main # this name is irrelevant.
+    template:
+      spec:
+        containers:
+        - name: job
+          image: hello-world
+          resources:
+            requests:
+              cpu: 1m
+```
+
+<<[UNRESOLVED creating a Workload beforehand]>>
+Could users create a Workload object before hand for groups of Pods?
+This way, Pods would only need to have a label for the group name and the role. This would
+simplify validation.
+Should we make this a supported mode in addition to pure labels/annotations?
+<<[/UNRESOLVED]>>
```

Comment:
Can we remove this? I think the `Users create a Workload object beforehand` section covers here.

Reply:
Oops, I thought I did already.